### PR TITLE
Add all/any tags filter command for customer

### PR DESCRIPTION
**Pull request by nvmdava**
allcus tag1 tag2 tag3... to find all customers that have all of the tags
anycus tag1 tag2 tag3... to find all customers that have at least of the tags

can @zlimez review and also add any/all tags for commissions?

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AllTagCustomerCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AllTagCustomerCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AllTagCustomerCommand.MESSAGE_USAGE));
+        }
+
+        String[] tagsAsString = trimmedArgs.split("\\s+");
```
Comment (zlimez) [inline src/main/java/seedu/address/logic/parser/AllTagCustomerCommandParser.java:30]: Hmmm, if I remember correctly tags can be multiple words right, so might need to either change the constraints of tags or modify how we parse here?

**Comment by zlimez** [comment]
Overall, LGTM to me, aside from my small concern on tag parsing hehe.

**Review by zlimez — APPROVED** [review]
I can see what I can do with the tags 😬

**Comment by ngxingyu** [comment]
The `addcus` and `editcus` already supports parsing for multiple tags so perhaps we can follow how it is done there. I don't think we should merge this until the tag parsing issue is fixed

**Pull request merged**